src/engine/sim_engine/pool.rs

```diff
@@ -364,27 +377,57 @@ impl Pool for SimPool {
             }
         };
 
-        if encryption_info.key_description().is_none() {
-            return Err(StratisError::Msg(
-                "This device is not bound to a keyring passphrase; refusing to remove \
-                the only unlocking method"
-                    .to_string(),
-            ));
-        }
+        let token_slot_to_add = match token_slot {
+            OptionalTokenSlotInput::Some(t) => {
+                if let Some(info) = encryption_info.get_info(t) {
+                    match info {
+                        UnlockMechanism::KeyDesc(ref key_desc) => {
+                            if key_description != key_desc {
+                                return Err(StratisError::Msg(format!(
+                                    "Key slot {t} is already in use by key description {}; requested {}",
+                                    key_desc.as_application_str(),
+                                    key_description.as_application_str(),
+                                )));
+                            } else {
+                                return Ok(CreateAction::Identity);
+                            }
+                        }
+                        UnlockMechanism::ClevisInfo(_) => {
+                            return Err(StratisError::Msg(format!(
+                                "Key slot {t} is already in use by a Clevis token"
+                            )));
+                        }
+                    }
+                } else {
+                    t
+                }
+            }
+            OptionalTokenSlotInput::None => encryption_info.free_token_slot(),
```

Comment:
Right here, it no token specified, a new token slot is found. To be consistent, since this is keyring, should check against existing encryption info and return Identity if already bound with this key.

Reply:
After looking into this, I remembered this was intentional and now recall why I made this decision. This would require reworking the API to pass the key store into this method. I don't think we should rework the API just for the sim_engine, but I'm open to other suggestions.